ras_backstage/resources/reporting_units/get_reporting_unit.py

```diff
@@ -62,6 +66,23 @@ def get(ru_ref):
         return make_response(jsonify(response_json), 200)
 
 
+def survey_ids_for_respondent(respondent, ru_ref):
+    return [enrolment.get('surveyId')
```

Comment:
I would split this into two list comprehensions
respondent_enrolments = [association.get('enrolments') for association in respondent.get('associations') if association['sampleUnitRef'] == ru_ref][0]]

return [enrolement.get('surveyId) for enrolment in respondent_enrolments]

The nested list comprehensions are a bit complicated.

Reply:
:+1: I agree with this
There are one or two places where we still have nested list comprehensions but usually they're pretty tightly bound together